### PR TITLE
Adding UT for Powerstore install support

### PR DESCRIPTION
**Pull request by panigs7**
# Description
Added UT for Powerstore install support

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/613 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Ran the unit test using go test
image

### CODE REVIEW

**Review by rajendraindukuri — DISMISSED** [review]

File: controllers/csm_controller_test.go [hunk 1]
```diff
@@ -226,6 +226,26 @@ func (suite *CSMControllerTestSuite) TestPowerFlexAnnotation() {
...
+func (suite *CSMControllerTestSuite) TestPowerStoreAnnotation() {
+
+	csm := shared.MakeCSM(csmName, suite.namespace, configVersion)
+	csm.Spec.Driver.Common.Image = "image"
+	csm.Spec.Driver.CSIDriverType = csmv1.PowerStore
+
+	csm.ObjectMeta.Finalizers = []string{CSMFinalizerName}
+
+	suite.fakeClient.Create(ctx, &csm)
+	sec := shared.MakeSecret(csmName+"-creds", suite.namespace, configVersion)
```
Comment (karthikk92) [inline controllers/csm_controller_test.go:238]: I think"-config" is the right keyword

File: tests/config/driverconfig/powerstore/v2.4.0/controller.yaml [hunk 2]
```diff
@@ -0,0 +1,314 @@
+apiVersion: v1
```
Comment (rensyct) [inline tests/config/driverconfig/powerstore/v2.4.0/controller.yaml:1]: @panigs7, do we need these files repeated under tests for each of the supported version. Can we use them from driverconfig if required and not have it repeated

File: controllers/csm_controller_test.go (hunk 1 above)
Comment (panigs7) [reply controllers/csm_controller_test.go]: Done

File: tests/config/driverconfig/powerstore/v2.4.0/controller.yaml (hunk 2 above)
Comment (panigs7) [reply tests/config/driverconfig/powerstore/v2.4.0/controller.yaml]: Discussed offline. Will be taking as an investigation

File: pkg/drivers/common_test.go [hunk 3]
```diff
@@ -135,3 +135,47 @@ func csmForPowerFlex(customCSMName string) csmv1.ContainerStorageModule {
...
+	// Add FSGroupPolicy
+	res.Spec.Driver.CSIDriverSpec.FSGroupPolicy = "File"
+
+	// Add DNS Policy for GetNode test
+	res.Spec.Driver.DNSPolicy = "ThisIsADNSPolicy"
+
+	// Add image name
+	res.Spec.Driver.Common.Image = "thisIsAnImage"
+
+	// Add pscale driver version
```
Comment (jooseppi-luna) [inline pkg/drivers/common_test.go:151]: pstore not pscale

File: pkg/drivers/common_test.go [hunk 4]
```diff
@@ -135,3 +135,47 @@ func csmForPowerFlex(customCSMName string) csmv1.ContainerStorageModule {
...
+	res.Spec.Driver.ConfigVersion = version
+
+	// Add pscale driver type
```
Comment (jooseppi-luna) [inline pkg/drivers/common_test.go:154]: pstore not pscale

**Review by jooseppi-luna — CHANGES_REQUESTED** [review]

**Comment by jooseppi-luna** [comment]
If we change the comments I'm ready to approve

File: pkg/drivers/common_test.go (hunk 3 above)
Comment (panigs7) [reply pkg/drivers/common_test.go]: Done

File: pkg/drivers/common_test.go (hunk 4 above)
Comment (panigs7) [reply pkg/drivers/common_test.go]: Done

**Comment by panigs7** [comment]
> If we change the comments I'm ready to approve

Address the review comments. Thanks @jooseppi-luna for reviewing!!

**Review by jooseppi-luna — APPROVED** [review]
lgtm

**Review by rensyct — APPROVED** [review]

**Review by karthikk92 — APPROVED** [review]

**Pull request merged**